### DISCUSSION TITLE
Is there a way to use regex for field columns in WHERE?

### DISCUSSION THREAD

**Discussion by aborruso** [nushell/nushell#8445] [Q&A]
Hi,
in SELECT I can use regex to select columns (`SELECT COLUMNS('number\d+') FROM addresses;`).

I have a table with more than 200 columns. I would like to run this kind of query

```
select  * from  input where COLUMNS('.+DATA.+') >'2022-10-10'
```

And to have in OR all the columns that match the regex.

Thank you

**Comment by fdncred** [comment]
I don't think so. We'd need to be able to do something like this.
```
let columns = ($table | columns) # This creates a list of all the columns
let first_three = $columns | first 3 
$table | select $first_three
```
There may be a way to do this in script but it's escaping me at the moment.
